test/receiver/receiver.go

```diff
@@ -65,13 +66,43 @@ func main() {
 	outDir := flag.String("outdir", "", "Output directory. Absolute path. Optional.")
 	profiler := flag.String("profiler", "", "Where should the profiler listen?")
 	exitAfter := flag.Duration("exitAfter", time.Second*10, "Exit after not receiving any message for this time. Will work only of outDir == ''. Will not exit if Duration is 0")
+	localAPIPortFlag := flag.Int64("local-api-port", 8024, "specify which port the local HTTP API should be listening on")
 
 	flag.Parse()
 
 	if *portsStr == "" {
 		log.Fatal("please supply the ports argument")
 	}
 
+	type status struct {
+		Ready        bool
+		IdleTimeSecs int64
+	}
+
+	var ready bool = false
```

Comment:
Should be atomic. Potential race.

Reply:
there is no consequence of having a race condition here I think. value is written once, in one place, and it's not a big deal if we read "false" whereas it just became "true". The opposite is never going to happen so we're good